### PR TITLE
CropSpec metadata can handle optional non String values

### PR DESCRIPTION
**Pull request by tonytw1**
## What does this change?

CropSpec metadata currently persists Option None as literal string "None" .
This does not look intentional and fails for optional non string values.

ie. The current code round trips CropSpec fields like so:

`Some("Hello")` -> `Some("Hello")`
`Some(1)` -> `Some(1)`
`None` -> `Some("None")`
`None[Int]` -> Fails with cannot parse `"None"` as an Int ❌ 

After this change, the following happens:

`Some("Hello")` -> `Some("Hello")`
`Some(1)` -> `Some(1)`
`None` -> `None`
`None[Int]` -> `None[Int]`

- Pull the serialization method into a new trait for testing.
- Change the code so that Option None round trips to None. More verbose implementation to spell out the steps.

This change is backwards compatible but makes no attempt to clean the existing data.
ie. optional None Strings which have already been persisted as `"None"` will continue to be read as Some("None").

This issue was noticed when trying to use an optional non String value in the CropSpec (ie. an Int field).

## Tested? Documented?
- [ ] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment
- [ ] relevant documentation added or amended (if needed)

### CODE REVIEW

**Comment by tonytw1** [comment]
Rebased.

File: cropper/app/lib/CropSpecMetadata.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    val nonEmptyMetadata = metadata.filter {
+      case (_, None) => false
+      case _ => true
+    }
+
+    val flattenedMetadata = nonEmptyMetadata.collect {
+      case (key, Some(value)) => key -> value
+      case (key, value) => key -> value
+    }.view.mapValues(_.toString).toMap
```
Comment (andrew-nowak) [inline cropper/app/lib/CropSpecMetadata.scala:30]:
```suggestion
    val flattenedMetadata = metadata.collect {
      case (key, Some(value))            => key -> value.toString
      case (key, value) if value != None => key -> value.toString
    }

    flattenedMetadata
```

**Review by andrew-nowak — APPROVED** [review]
looks great! I've got just one suggestion to simplify slightly

**Comment by andrew-nowak** [comment]
Thanks @tonytw1 !

**Pull request merged**